src/rp2040_flash.rs

```diff
@@ -42,3 +45,43 @@ pub fn read_device_config_from_rp2040_flash() -> &'static [u8] {
     let addr = (FLASH_XIP_BASE + FLASH_END - FLASH_USER_SIZE) as *const u8;
     unsafe { slice::from_raw_parts(addr, FLASH_USER_SIZE as usize) }
 }
+
+pub fn read_is_audio_from_rp2040_flash() -> bool {
+    let addr = (FLASH_XIP_BASE + FLASH_END - FLASH_USER_SIZE) as *const u8;
+    let config = unsafe { slice::from_raw_parts(addr, 5 as usize) };
+    if config[0] == u8::MAX && config[1] == u8::MAX && config[2] == u8::MAX && config[3] == u8::MAX
+    {
+        false
+    } else {
+        config[4] == 1
+    }
+}
+
+pub fn read_alarm_from_rp2040_flash() -> &'static [u8] {
+    let mut addr = FLASH_XIP_BASE + FLASH_END;
+    if addr % 256 != 0 {
+        addr = 256 * (1 + (addr / 256) as u32);
+    }
+    unsafe { slice::from_raw_parts(addr as *const u8, 256 as usize) }
```

Comment:
Small style nit, rather than `256 as usize`, it's customary to just write `256usize`.